sklearn/tests/test_common.py

```diff
@@ -624,3 +625,27 @@ def test_set_output_transform_configured(estimator, check_func):
     _set_checking_parameters(estimator)
     with ignore_warnings(category=(FutureWarning)):
         check_func(estimator.__class__.__name__, estimator)
+
+
+@pytest.mark.parametrize(
+    "estimator", _tested_estimators(), ids=_get_check_estimator_ids
+)
+def test_check_inplace_ensure_writeable(estimator):
+    if hasattr(estimator, "copy"):
+        estimator.set_params(copy=False)
+    elif hasattr(estimator, "copy_X"):
+        estimator.set_params(copy_X=False)
+    else:
+        raise SkipTest("Estimator doesn't require writeable input.")
+
+    _set_checking_parameters(estimator)
+
+    # The following estimators can work inplace only with certain settings
+    if estimator.__class__.__name__ == "HDBSCAN":
```

Comment:
Given that we are in our testing code `sklearn/tests/test_common.py` can we use `isinstance` here?

Reply:
I used the class name because it's what we use in `_set_checking_parameters(estimator)`. Otherwise we have to import all all estimators for which we want to set a specific param which feels a bit cumbersome.